store.go

```diff
@@ -1056,11 +1057,14 @@ func (s *store) imageTopLayerForMapping(image *Image, ristore ROImageStore, crea
 				// We want the layer's parent, too, if it has one.
 				var cParentLayer *Layer
 				if cLayer.Parent != "" {
-					// Its parent should be around here, somewhere.
-					if cParentLayer, err = store.Get(cLayer.Parent); err != nil {
-						// Nope, couldn't find it.  We're not going to be able
-						// to diff this one properly.
-						continue
+					// Its parent should be in one of the stores, somewhere.
+					for _, ps := range allStores {
+						if cParentLayer, err = ps.Get(cLayer.Parent); err != nil {
+							// Nope, couldn't find it.  We're not going to be able
+							// to diff this one properly.
+							continue
```

Comment:
This `continue` now skips to the next iteration of the loop started at line 1061, instead of the loop that starts at line 1055.

Reply:
Fixed, now it sets a found flag when it find the parent layer, otherwise when this loop finishes it checks it loops if not found.